src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -0,0 +1,568 @@
+    # if we need to update the tasks, we will update the cron expression from it
+    # if not we just update the cron expression from the given parameter
+    acr_task_client = cf_acr_tasks(cmd.cli_ctx)
+    for task in task_list:
+        deployed_task = task.step.encoded_task_content
+        extension_task = _create_encoded_task(CONTINUOUSPATCH_TASK_DEFINITION[task.name]["template_file"])
+        if deployed_task != extension_task:
+            logger.debug(f"Task {task.name} is different from the extension task, updating the task")
+            _update_task_yaml(acr_task_client, registry, resource_group, task, extension_task)
+
+    if schedule_cron_expression is not None:
+        _update_task_schedule(acr_task_client, registry, resource_group, schedule_cron_expression, dry_run)
+
+
+def _eval_trigger_run(cmd, registry, resource_group, run_immediately):
+    if run_immediately:
+        logger.warning(f'Triggering the {CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME} to run immediately')
+        # Seen Managed Identity taking time, see if there can be an alternative (one alternative is to schedule the cron expression with delay)
+        # NEED TO SKIP THE TIME.SLEEP IN UNIT TEST CASE OR FIND AN ALTERNATIVE SOLUITION TO MI COMPLETE
+        time.sleep(30)
+        _trigger_task_run(cmd, registry, resource_group, CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME)
+
+
+def delete_continuous_patch_v1(cmd, registry, dryrun):
+    logger.debug("Entering delete_continuous_patch_v1")
+    cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
+    cssc_config_exists = check_continuous_task_config_exists(cmd, registry)
+    if not dryrun and (cssc_tasks_exists or cssc_config_exists):
+        cssc_tasks = ', '.join(CONTINUOUSPATCH_ALL_TASK_NAMES)
+        logger.warning(f"All of these tasks will be deleted: {cssc_tasks}")
+        for taskname in CONTINUOUSPATCH_ALL_TASK_NAMES:
+            # bug: if one of the deletion fails, the others will not be attempted, we need to attempt to delete all of them
+            _delete_task(cmd, registry, taskname, dryrun)
+            logger.warning(f"Task {taskname} deleted.")
+        logger.warning(f"Deleting {CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}:{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1}")
+        delete_oci_artifact_continuous_patch(cmd, registry, dryrun)
+
+    if not cssc_tasks_exists:
+        logger.warning(f"{ERROR_MESSAGE_WORKFLOW_TASKS_DOES_NOT_EXIST}")        
+
+def list_continuous_patch_v1(cmd, registry):
+    logger.debug("Entering list_continuous_patch_v1")
+    cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
+    if not cssc_tasks_exists:
+        logger.warning(f"{ERROR_MESSAGE_WORKFLOW_TASKS_DOES_NOT_EXIST}")
+        return
+
+    acr_task_client = cf_acr_tasks(cmd.cli_ctx)
+    resource_group_name = parse_resource_id(registry.id)[RESOURCE_GROUP]
+    tasks_list = acr_task_client.list(resource_group_name, registry.name)
```

Comment:
Do you fetch all tasks from the server and then filter out the CSSC tasks? This could lead to performance issues if there are many tasks in the registry. Could you fetch only the CSSC-specific tasks to improve performance?

Reply:
changed call from 'list()' to 'get()' using the CSSC task names